perfkitbenchmarker/linux_virtual_machine.py

```diff
@@ -708,3 +731,137 @@ def SetupProxy(self):
+    self.RemoteHostCommand('mkdir -p %s' % vm_util.VM_TMP_DIR)
+    self.Install('docker')
+    self.InitDocker()
+    super(ContainerizedDebianMixin, self).PrepareVMEnvironment()
+
+  def InitDocker(self):
+    """Initializes the docker container daemon."""
+    init_docker_cmd = ['sudo docker run -d '
+                       '--net=host '
+                       '--workdir=%s '
+                       '-v %s:%s ' % (CONTAINER_WORK_DIR,
+                                      vm_util.VM_TMP_DIR,
+                                      CONTAINER_MOUNT_DIR)]
+    for sd in self.scratch_disks:
+      init_docker_cmd.append('-v %s:%s ' % (sd.mount_point, sd.mount_point))
+    init_docker_cmd.append('%s sleep infinity ' % UBUNTU_DOCKER_IMAGE)
+    init_docker_cmd = ''.join(init_docker_cmd)
+
+    resp, _ = self.RemoteHostCommand(init_docker_cmd)
+    self.docker_id = resp[:-1]
+    return self.docker_id
+
+  def RemoteCommand(self, command,
+                    should_log=False, retries=SSH_RETRIES,
+                    ignore_failure=False, login_shell=False,
+                    suppress_warning=False):
+    """Runs a command inside the container.
+
+    Args:
+      command: A valid bash command.
+      should_log: A boolean indicating whether the command result should be
+          logged at the info level. Even if it is false, the results will
+          still be logged at the debug level.
+      retries: The maximum number of times RemoteCommand should retry SSHing
+          when it receives a 255 return code.
+      ignore_failure: Ignore any failure if set to true.
+      login_shell: Run command in a login shell.
+      suppress_warning: Suppress the result logging from IssueCommand when the
+          return code is non-zero.
+
+    Returns:
+      A tuple of stdout and stderr from running the command.
+    """
+    # Escapes bash sequences
+    command = command.replace("\'", "\'\\\'\'")
+
+    logging.info('Docker running: %s' % command)
+    command = "sudo docker exec %s bash -c '%s'" % (self.docker_id, command)
+    return self.RemoteHostCommand(command, should_log, retries,
+                                  ignore_failure, login_shell, suppress_warning)
+
+  def ContainerCopy(self, file_name, container_path='', copy_to=True):
+    """Copies a file to and from container_path to the host's vm_util.VM_TMP_DIR.
+
+    Args:
+      file_name: Name of the file in the host's vm_util.VM_TMP_DIR.
+      container_path: Optional path of where to copy file on container.
+      copy_to: True to copy to container, False to copy from container.
+    Raises:
+      RemoteExceptionError: If the source container_path is blank.
+    """
+    if copy_to:
+      if container_path == '':
+        container_path = CONTAINER_WORK_DIR
+
+      # Everything in vm_util.VM_TMP_DIR is directly accessible
+      # both in the host and in the container
+      source_path = os.path.join(CONTAINER_MOUNT_DIR, file_name)
```

Comment:
I believe we have a bigger problem here. Many benchmarks use os.path, so ContainerizedVM running on a Windows machine would break for those benchmarks for the same reason, right?